cmake/modules/openmp.cmake

```diff
@@ -1,5 +1,19 @@
 option(USE_OpenMP "Use OpenMP" ON)
 
+# We don't want GOMP because its performance sinks for large core count, so we force libomp
+# This finds the library path from the system's clang for OpenMP
+#
+# On Fedora, it's at the same place as others, so we don't need to look elsewhere
+# On Ubuntu, it's in /usr/lib/llvm-${version}, so find_package finds GOMP for GCC instead.
+execute_process (
+    COMMAND bash -c "for lib in $(clang -lomp -### 2>&1); do echo $lib | grep -o \"\\/.*llvm.*\\w\"; done"
```

Comment:
It seem to not work as expected.

I tired running `for lib in $(clang -lomp -### 2>&1); do echo $lib | grep -o \"\\/.*llvm.*\\w\"; done` on both Ubuntu 22.04 and 24.04. I'm trying in docker: `docker run -it ubuntu:22.04 /bin/bash`. The only thing I do is `apt update; apt install clang libomp-dev`

On Ubuntu 22.04, I see that `clang -lomp -###` has `"-L/usr/lib/llvm-14/bin/../lib"` in it's output, but grep still doesn't catch it. On Ubuntu 24.04 there are no directories with `llvm` in it in clang's output.

Reply:
Thanks for checking!

My idea here is that, if `libomp` is in a "standard" directory (ex. `/lib64`), we don't need to add the extra library path. But if it's in an "LLVM specific" directory, we do. So, if `clang` adds any "LLVM specific" directory, we do, too.

So, even if the command line above doesn't find anything, it's still possible that we can continue to build with GCC as hoped.

Reply:
@dbabokin Just noticed, you quoted the grep when you ran (`\"` ), so your grep didn't _"catch"_ because it was broken. I just did this myself, too. :laughing: 

I'm running this on Ubuntu 22.04 and when I unquote the quote (as Bash does via CMake), it does catch the directory and the build works (where it previously didn't).